src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -125,6 +162,20 @@ private static void initHttpClient() {
     httpClient = clientBuilder.build();
   }
 
+  /** Starts a daemon thread to monitor idle connections in http connection manager. */
+  private static void initIdleConnectionMonitoringThread() {
```

Comment:
IIRC this function was causing issue when we looked at the Fidelity case, right?

Reply:
Correct but there are no strong evidence apart from the stack trace, both confluent and I tested this commit on my local machine(with extensive testing) and didnt see any issue.. We will ofcourse wait for this version to be tested with KC before releasing KC. 

But IMO there is nothing in this function, the tests didnt catch this either..

Probably result of more threads? (Cant tell for sure)

Reply:
But, let me wrap this around try catch because  I vaguely remember this was the line
```idleConnectionMonitorThread.start();```

Reply:
```
040ac5fef93977329|pr102703|pr102703 {"@timestamp":"2022-05-11T18:33:32.931+00:00","message":"Wo rkerSinkTask{id=msm_fcm_snowflake_connector_prod-12} Task threw an uncaught and unrecoverable exception","logger_name":"org.apache.kafka.connect.runtime.WorkerTask","thread_name":"task-t hread-msm_fcm_snowflake_connector_prod-12","level":"ERROR","stack_trace":"com.snowflake.kafka.connector.internal.SnowflakeKafkaConnectorException: 
[SF_KAFKA_CONNECTOR] Exception: Inval id private key
[SF_KAFKA_CONNECTOR] Error Code: 0002
[SF_KAFKA_CONNECTOR] Detail: private key should be a valid PEM RSA private key
[SF_KAFKA_CONNECTOR] Message: null
[SF_KAFKA_CONNECTOR] java.lang.Thread.start(Thread.java:710)
[SF_KAFKA_CONNECTOR] net.snowflake.ingest.utils.HttpUtil.initIdleConnectionMonitoringThread(HttpUtil.java:173)
[SF_KAFKA_CONNECTOR] net.sn owflake.ingest.utils.HttpUtil.getHttpClient(HttpUtil.java:77)
[SF_KAFKA_CONNECTOR] net.snowflake.ingest.SimpleIngestManager.init(SimpleIngestManager.java:409)
[SF_KAFKA_CONNECTOR] net. snowflake.ingest.SimpleIngestManager.<init>(SimpleIngestManager.java:383)
[SF_KAFKA_CONNECTOR] com.snowflake.kafka.connector.internal.SnowflakeIngestionServiceV1.<init>(SnowflakeIngesti onServiceV1.java:46)
[SF_KAFKA_CONNECTOR] com.snowflake.kafka.connector.internal.SnowflakeIngestionServiceFactory$SnowflakeIngestionServiceBuilder.<init>(SnowflakeIngestionServiceFactor y.java:44)
[SF_KAFKA_CONNECTOR] com.snowflake.kafka.connector.internal.SnowflakeIngestionServiceFactory$SnowflakeIngestionServiceBuilder.<init>(SnowflakeIngestionServiceFactory.java:31) 
```

Reply:
Yes, we want to return the actual exception so we know what happened next time

Reply:
Almost certain that this is because of confluent version + threads and not our code. 

I tested this with 200 partition and continuously restarting the tasks to hit the close and open API.

Reply:
This is fixed in https://github.com/snowflakedb/snowflake-ingest-java/commit/de681c650f5ac221c5b13af34771c1fbb6eaec12